### ISSUE TITLE
"parser_frontends.py:216: FutureWarning: Possible nested set" when using nested sets and regex

### ISSUE THREAD

**Issue by wjl** [lark-parser/lark#897]
When using lark with the "regex" library, which supports nested sets, lark reports a FutureWarning about nested sets, which seems to indicate that it is passing the pattern to the "re" library at some point. However, besides the warning, the pattern with the nested sets (and other "regex"-specific features) work fine, so it is for sure actually using the "regex" library during actual exection.

Here is a quick way to reproduce:

```
$ cd /tmp
$ python -m venv venv
$ . ./venv/bin/activate
$ pip install 'lark-parser[regex]'
$ cat <<EOF > use-nested-set.py
#!/bin/env python

import lark
import regex

grammar = """
start: nested_set
nested_set: /[[\p{XID_Start}_\N{OVERLINE}]--[\p{Other_ID_Start}]]+/
"""

parser = lark.Lark(
	grammar=grammar
	,parser="earley"
	,lexer="dynamic"
	,ambiguity="resolve"
	,start="start"
	,debug=False
	,regex=True
	,g_regex_flags=(0
		| regex.FULLCASE
		| regex.MULTILINE
		| regex.UNICODE
		| regex.VERBOSE
		| regex.VERSION1
		| regex.WORD
	)
	,maybe_placeholders=True
	,propagate_positions=True
)

text = "hello"
tree = parser.parse(text)
print(tree.pretty())
EOF
$ python use-nested-sets.py
/home/wjl/tmp/venv/lib/python3.9/site-packages/lark/parser_frontends.py:216: FutureWarning: Possible nested set at position 1
  return f(lexer_conf, parser_conf, options, resolve_ambiguity=resolve_ambiguity, debug=debug, tree_class=tree_class, **extra)
start
  nested_set	hello
```

[use-nested-sets.py.txt](https://github.com/lark-parser/lark/files/6513058/use-nested-sets.py.txt)

**Comment by MegaIng** [comment]
This Warning is not generated directly by `re`, but by `sre_parse` which we use the get min and max length of strings. That module ddecides 'noone will ever call this from outside `re`' and decides to create the warning 6 levels above itself, which points somewhere random in our Code base. There is not really a fix here, unless we decide to not use `sre_parse` when `regex=True`.

**Comment by wjl** [comment]
Well, sounds like you are aware of the issue. I am not sure exactly why you are using sre_parse; I use regexes with chart parsers all the time and never have to do any preprocessing on them. But I'm sure you have a good reason based on how you implement things. In any case, I would be great if you could fully support regex.

When I use "regex=True" I really want to be able to use all regex features, not just a small subset. This (and related issues, like sre_parse throwing fatal errors instead of just warnings on some regex syntax) basically makes regex=True pretty much unusable for me for now. I realize I can work around this by making a separate lexer but that basically defeats my entire reason for trying to use lark.

Just some context:

- I write a lot of DSLs; usually I use ad-hoc recursive descent or a hand-rolled earley hybrid chart parser.
- Everyone once in a while, I try to use lark on a project instead because it is more polished than my code in most respects.
- If I hit a major issue (like regex support not really working) I could work around it, but usually I go back to other solutions.
- I only ever use regex (with the V1 flag), because it is the only sane regex engine available in Python for industrial use.
- I only ever use generalized parsing in Python; in situations where I want something faster, I use RD and don't use Python.
- As a user, I'm already installing lark with pip. I'm not sure why I would care if it had a 100% dependency on regex.
- This is especially true, given that other modules I already use pull in regex anyway.

Anyway, hope this helps, and I look forward to being able to use regex 100% in lark some time in the future! =)

**Comment by MegaIng** [comment]
@wjl You can manually apply #878 . It adds a fallback that should prevent the function from completly erroring out. Warnings will still be generated.

We are using `sre_parse` to get the minmal and maximal width of a regex. We need this information to correctly behave with in the lexer. Sadly, there are no equivalent functions for the `regex` module, so we try to use the standard module. As you noticed this breaks when it hits unsupported syntax.

It might be worth sending a ticket to the `regex` asking them to add a corresponding method, but that might take a while. In the mean time, @erezsh I would suggest at least adding the fallback, potentially even with a warning suppression to close this issue.

**Comment by erezsh** [comment]
@MegaIng 

I merged #878 to `master`.

We might consider to entirely skip sre_parse when `regex=True`. It means slightly worse automatic disambiguation with LALR, but if a user is already doing lalr=True *and* regex=True, they can probably handle setting an explicit priority where necessary.

**Comment by erezsh** [comment]
You may re-open if this is still an issue.
